src/otx/cli/utils/installation.py

```diff
@@ -0,0 +1,518 @@
+
+
+def get_cuda_version() -> str | None:
+    """Get CUDA version installed on the system.
+
+    Examples:
+        >>> # Assume that CUDA version is 11.2
+        >>> get_cuda_version()
+        "11.2"
+
+        >>> # Assume that CUDA is not installed on the system
+        >>> get_cuda_version()
+        None
+
+    Returns:
+        str | None: CUDA version installed on the system.
+    """
+    # 1. Check CUDA_HOME Environment variable
+    cuda_home = os.environ.get("CUDA_HOME", "/usr/local/cuda")
+
+    if Path(cuda_home).exists():
+        # Check $CUDA_HOME/version.json file.
+        version_file = Path(cuda_home) / "version.json"
+        if not version_file.is_file():
+            return None
+        with Path(version_file).open() as file:
+            data = json.load(file)
+            cuda_version = data.get("cuda", {}).get("version", None)
+            if cuda_version is not None:
+                cuda_version_parts = cuda_version.split(".")
+                return ".".join(cuda_version_parts[:2])
+    else:
+        # 2. 'nvcc --version' check
+        try:
+            result = subprocess.run(args=["nvcc", "--version"], capture_output=True, text=True, check=False)
+            output = result.stdout
+
+            cuda_version_pattern = r"cuda_(\d+\.\d+)"
+            cuda_version_match = re.search(cuda_version_pattern, output)
+
+            if cuda_version_match is not None:
+                return cuda_version_match.group(1)
+        except Exception:
+            msg = "Could not find cuda-version. Instead, the CPU version of torch will be installed."
+            warn(msg, stacklevel=2)
+    return None
+
+
+def update_cuda_version_with_available_torch_cuda_build(cuda_version: str, torch_version: str) -> str:
+    """Update the installed CUDA version with the highest supported CUDA version by PyTorch.
+
+    Args:
+        cuda_version (str): The installed CUDA version.
+        torch_version (str): The PyTorch version.
+
+    Raises:
+        Warning: If the installed CUDA version is not supported by PyTorch.
+
+    Examples:
+        >>> update_cuda_version_with_available_torch_cuda_builds("11.1", "1.13.0")
+        "11.6"
+
+        >>> update_cuda_version_with_available_torch_cuda_builds("11.7", "1.13.0")
+        "11.7"
+
+        >>> update_cuda_version_with_available_torch_cuda_builds("11.8", "1.13.0")
+        "11.7"
+
+        >>> update_cuda_version_with_available_torch_cuda_builds("12.1", "2.0.1")
+        "11.8"
+
+    Returns:
+        str: The updated CUDA version.
+    """
+    selected_cuda_version = None
+    max_supported_cuda = max(AVAILABLE_TORCH_VERSIONS[torch_version]["cuda"])
+    min_supported_cuda = min(AVAILABLE_TORCH_VERSIONS[torch_version]["cuda"])
+
+    if cuda_version > max_supported_cuda:
+        selected_cuda_version = max_supported_cuda
+    elif cuda_version < min_supported_cuda:
+        selected_cuda_version = min_supported_cuda
+
+    if selected_cuda_version is not None:
```

Comment:
```suggestion
    max_supported_cuda = max(AVAILABLE_TORCH_VERSIONS[torch_version]["cuda"])
    min_supported_cuda = min(AVAILABLE_TORCH_VERSIONS[torch_version]["cuda"])

    bounded_cuda_version = max(min(cuda_version, max_supported_cuda), min_supported_cuda)
    
    if cuda_version != selected_cuda_version:
```

Sorry that this might bother you. But, please understand me. I'm really beware of bloating `if-else` statements without cautions in the codebase like OTX v1. Especially, this is because we are very early phase of development.

Reply:
No problem. https://github.com/openvinotoolkit/training_extensions/pull/2679/commits/b53e0a9e597829cd0a54a3c76c26026c352f2a29
-> I modified it as your suggestion.